experimental/IntersectionTheory/src/blowup.jl

```diff
@@ -129,58 +166,53 @@ function  present_finite_extension_ring(F::Oscar.AffAlgHom)
   end
   if B isa MPolyQuoRing
     BR = base_ring(B)
-    M = [F(gens(A)[i]).f for i = 1:ngens(A)]
+    M = [F(gens(A)[i]).f for i = 1:a]
   else
     BR = B
-    M = [F(gens(A)[i]) for i = 1:ngens(A)]
+    M = [F(gens(A)[i]) for i = 1:a]
   end
+
   @assert base_ring(AR) == base_ring(BR)
 
   I = ideal(BR, isdefined(B, :I) ? vcat(gens(B.I), M) : M)
   C, _ = quo(BR, I)
-  gensB = monomial_basis(C)
-  @assert gensB[end] == 1 # the last one should always be 1
-  g = length(gensB)
+  gs = monomial_basis(C) # monomials whose residue classes form a K-basis of B
+  @assert gs[end] == 1 # the last one should always be 1
+  g = length(gs)
 
-  base = base_ring(AR)
-  symsA, symsB = string.(gens(AR)), string.(gens(BR))
-  a, b = length(symsA), length(symsB)
   R, _ = tensor_product(BR, AR, use_product_ordering = true)
   ba = gens(R)
-
-  AtoR = hom(AR, R, ba[b+1:end], check = false)
-  BtoR = hom(BR, R, ba[1:b], check = false)
-  RtoA = hom(R, AR, vcat(repeat([AR()], b), gens(AR)))
-  
-  gensB_lift = [R(BtoR(g)) for g in gensB]
+  ARtoR = hom(AR, R, ba[b+1:end], check = false)
+  BRtoR = hom(BR, R, ba[1:b], check = false)
+  RtoAR = hom(R, AR, vcat(repeat([AR()], b), gens(AR)))
+  gs_lift = [BRtoR(g) for g in gs]
   
   # compute the ideal J of the graph of F
-  rels = [R(ba[b+i]-BtoR(m)) for (i,m) in enumerate(M)]
-  if isdefined(A, :I) for g in gens(A.I) push!(rels, R(AtoR(g))) end end
-  if isdefined(B, :I) for g in gens(B.I) push!(rels, R(BtoR(g))) end end
-  J = ideal(R, rels) # the ideal of the graph of F
-
+  Rels = [ba[b+i]-BRtoR(m) for (i,m) in enumerate(M)]
+  if isdefined(A, :I) for g in gens(A.I) push!(Rels, ARtoR(g)) end end
+  if isdefined(B, :I) for g in gens(B.I) push!(Rels, BRtoR(g)) end end
+  J = ideal(R, Rels) # the ideal of the graph of F
   V = groebner_basis(J)
-  pf = x -> (y = reduce(R(BtoR(BR(x))), gens(V));
-	     ans = Nemo.elem_type(AR)[];
-	     for i in 1:g
-	       q = div(y, gensB_lift[i])
-	       push!(ans, RtoA(q))
-	       y -= q * gensB_lift[i]
-	     end; ans)
+
+  sect = x -> (y = reduce(BRtoR(x), gens(V));
+	      ans = elem_type(AR)[];
+	      for i in 1:g
+	        q = div(y, gs_lift[i])
+	        push!(ans, RtoAR(q))
+	        y -= q * gs_lift[i]
+	      end; ans)
```

Comment:
there are some more tabs in the above paragraph

Reply:
@lgoettgens please feel free to remove the tabs

Reply:
done (in https://github.com/oscar-system/Oscar.jl/pull/4180/commits/7435872d4eae7b0e667537b5c917493da60f945f)